flowise-embed/src/features/menu/components/Menu.tsx

```diff
@@ -1,26 +1,183 @@
+      await documentService.deleteChat(selectedChatId() as string);
+      fetchChatIds();
+    }
+  };
+
+  const getChatHistoryTitle = () => {
+    if (currentFlow() === 'compliance') {
+      return 'Análise de Compliance';
+    } else if (currentFlow() === 'critical_analysis') {
+      return 'Análise Crítica';
+    } else {
+      return 'Estimativa de Custos';
+    }
+  };
+
+  const formatDateChat = (item: ChatItem) => {
+    if (item.chat_name) {
+      return item.chat_name;
+    } else {
+      const date = new Date(item.created_at);
+      return `Sem título - ${date.toLocaleDateString()} ${date.toLocaleTimeString()}`;
+    }
+  };
+
+  const preprocessChatItems = (chatItems: ChatItem[]) => {
+    const groups = {
+      today: [] as ChatItem[],
+      yesterday: [] as ChatItem[],
+      lastWeek: [] as ChatItem[],
+      lastMonth: [] as ChatItem[],
+      older: [] as ChatItem[],
+    };
+    const labels = {
+      today: 'Hoje',
+      yesterday: 'Ontem',
+      lastWeek: 'Últimos 7 dias',
+      lastMonth: 'Últimos 30 dias',
+      older: '30 dias ou mais atrás',
+    };
+
+    const adjustToBrazilTime = (date: Date) => {
+      // Manual adjustment for Brazil time zone (UTC-3)
+      const offset = -3 * 60; // UTC-3 in minutes
+      const utcDate = new Date(date.getTime() + date.getTimezoneOffset() * 60000);
+      return new Date(utcDate.getTime() + offset * 60000);
+    };
+
+    // Sort chatItems by created_at in descending order
+    chatItems.sort((a, b) => new Date(b.created_at).getTime() - new Date(a.created_at).getTime());
+
+    chatItems.forEach((item) => {
+      const date = adjustToBrazilTime(new Date(item.created_at));
+      const now = adjustToBrazilTime(new Date());
+
+      // Zero out hours, minutes, seconds, and milliseconds for day comparison
+      const dateOnly = new Date(date.getFullYear(), date.getMonth(), date.getDate());
+      const nowOnly = new Date(now.getFullYear(), now.getMonth(), now.getDate());
+
+      const diffTime = Math.abs(nowOnly.getTime() - dateOnly.getTime());
+      const diffDays = Math.ceil(diffTime / (1000 * 60 * 60 * 24));
+
+      if (diffDays === 0) {
+        groups.today.push(item);
+      } else if (diffDays === 1) {
+        groups.yesterday.push(item);
+      } else if (diffDays <= 7) {
+        groups.lastWeek.push(item);
+      } else if (diffDays <= 30) {
+        groups.lastMonth.push(item);
+      } else {
+        groups.older.push(item);
+      }
+    });
+
+    return { groups, labels };
+  };
+
+  const fetchChatIds = async () => {
+    const data = await documentService.getChatIdsByFlow(currentFlow());
+    if (data && Array.isArray(data)) {
+      setGroupedChatItems(preprocessChatItems(data as ChatItem[]));
+    }
+  };
+
+  const handleEditSubmit = async (event: Event, chatId: string) => {
+    event.preventDefault();
+    if (chatId.length > 0) {
+      await documentService.updateChatName(chatId, inputValue());
+      fetchChatIds();
```

Comment:
Há a necessidade de fazer o fetch novamente toda vez que o nome de um chat for atualizado?